src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,46 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        System.out.println(width + " -- " + length + " -- " + color);
+
+        if (this == object) {
+            return true;
+        }
+
+        if (null == object) {
+            return false;
+        }
+
+        if (object instanceof Rectangle) {
+            Rectangle rectangle = (Rectangle) object;
+            return checkNullOrEquals(width, rectangle.getWidth())
```

Comment:
let's not create other private methods for equals(). In this case it's better to have it in your method

Reply:
Done